src/interface/pipe.rs

```diff
@@ -58,6 +58,29 @@ impl EmulatedPipe {
         if (flags & O_RDWRFLAGS) == O_RDONLY {self.refcount_read.fetch_sub(1, Ordering::Relaxed);}
         if (flags & O_RDWRFLAGS) == O_WRONLY {self.refcount_write.fetch_sub(1, Ordering::Relaxed);}
     }
+    pub fn check_select_read(&self) -> bool {
+        let read_end = self.read_end.lock();
+        let pipe_space = read_end.len();
+
+        if (pipe_space > 0) && !self.eof.load(Ordering::SeqCst){
```

Comment:
Why do we check EOF here?
How can there be stuff in the read end of the pipe if we've already reached an EOF.
If the EOF is after the stuff in the read end, then select should still exit.

Reply:
I believe this actually should be an OR and just a eof load (non-negated).

One the EOF is set, select will always say the pipe is readable, even if theres nothing in it. But before its set, its only readable when there is data in the pipe.

Reply:
Noted.